Viral Pipeline Analysis Project
==============================
A project based analysis using the Artic Fieldbioinformatics Pipeline for viral pathogen analysis.

Contents
========

* [About the Artic Pipeline](#about-the-artic-pipeline)
* [Installation](#installation)
* [Usage](#usage)
* [Getting Started](#getting-started)
      * [Data Preparation](#data-preparation)

About the Artic Pipeline
========================
artic is a pipeline and set of accompanying tools for working with viral nanopore sequencing data, generated from tiling amplicon schemes.

It is designed to help run the artic bioinformatics protocols; for example the SARS-CoV-2 coronavirus protocol.

Features include:
* read filtering
* primer trimming
* amplicon coverage normalisation
* variant calling
* consensus building

There are 2 workflows baked into this pipeline, one which uses signal data (via nanopolish) and one that does not (via medaka).

Installation
============
Guidance on the installation of the Pipeline can be found at the author's github page: https://github.com/artic-network/fieldbioinformatics

Usage
=====
Documentation of the pipeline can be found at: http://artic.readthedocs.io/en/latest/?badge=latest

Getting Started
===============

### Data Preparation
